app/src/lib/shells/win32.ts

```diff
@@ -213,6 +240,64 @@ async function findGitBash(): Promise<string | null> {
   return null
 }
 
+/**
+ * Check if WSL is installed. When using this function also check the executable exists on the disk.
+ */
+async function isWslInstalled(): Promise<boolean> {
+  // First, determine that we are running Windows 10
+  const winVersion = OS.release() // Of the form '10.x.x'
+
+  if (!winVersion.startsWith('10.')) {
+    return false
+  }
+
+  // Second, check if WSL enabled?
+  const registryPath = enumerateValues(
+    HKEY.HKEY_LOCAL_MACHINE,
+    'SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Component Based Servicing\\Notifications\\OptionalFeatures\\Microsoft-Windows-Subsystem-Linux'
+  )
+
+  // TODO - does this check need to be more specific?
```

Comment:
Not sure if just the presence of the registry key is enough to confirm it's been installed?

Reply:
I think this is fine - when the key doesn't exist, you'll get an empty array.